awx/main/tasks.py

```diff
@@ -1234,6 +1243,10 @@ def run(self, pk, **kwargs):
                 status = 'failed'
                 extra_update_fields['job_explanation'] = self.instance.job_explanation
 
+        except InvalidVenvError as e:
+            extra_update_fields['job_explanation'] = e.message
+            logger.error('{} did not start because its custom venv is missing'.format(
```

Comment:
"is missing" isn't always accurate.  Couldn't we just use the message from the exception?

`logger.exception("{} did not start".format(self.instance.log_format))`

Reply:
I think this is because of the advice I gave. I don't think `logger.error` is wrong, because this is not a programming error, the traceback doesn't provide any additional meaningful information, as opposed to just the message. In fact, you could defer to the message, like `logger.error('{} {}'.format(self.instance.log_format, e.message))`, you're basically writing the same thing you wrote in the job record into the logs.

Reply:
`logger.error('{} {}'.format(self.instance.log_format, e.message))`

That'd work fine for me - I just think with the current version we're sort of repeating ourselves.

Reply:
@ryanpetrello @AlanCoding can we do

```logger.error('{} did not start because its custom venv is not discoverable'.format(self.instance.log_format))```

Reply:
Nevermind, going with the first one agreed upon :smile: